### DISCUSSION TITLE
How do I disable logging?

### DISCUSSION THREAD

**Discussion by AlyoshaVasilieva** [searxng/searxng#2021] [Q&A] [answered]
When I look at the log file `/var/log/uwsgi/app/searxng.log` I see entries like this:

```
Thu Dec 15 05:43:09 2022 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /autocompleter?q=pytho (ip 127.0.0.1) !!!
Thu Dec 15 05:43:09 2022 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /autocompleter?q=pytho (127.0.0.1)
OSError: write error
Thu Dec 15 05:43:10 2022 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /autocompleter?q=python.o (ip 127.0.0.1) !!!
Thu Dec 15 05:43:10 2022 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /autocompleter?q=python.o (127.0.0.1)
OSError: write error
Thu Dec 15 05:43:10 2022 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /autocompleter?q=python.org (ip 127.0.0.1) !!!
Thu Dec 15 05:43:10 2022 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /autocompleter?q=python.org (127.0.0.1)
OSError: write error
```

which shows what I'm typing into my browser's URL bar. It also shows search queries performed by GET if I close the tab while they're in process. How do I turn this off?

**Comment by return42** [comment, marked as answer]
How did you installed SearXNG, what distro do you use?  / [How uWSGI is implemented by distributors varies.](https://docs.searxng.org/admin/installation-uwsgi.html#origin-uwsgi)

If you use docker see:

https://github.com/searxng/searxng/blob/9e9f57e48bbc3c07d49dad57bb851f59f1c7def8/dockerfiles/uwsgi.ini#L30-L32

If you installed by the [scripts](https://docs.searxng.org/admin/installation-scripts.html) (or the "step by step" manual) the logging should already be disabled in the [uWSGI setup](https://docs.searxng.org/admin/installation-uwsgi.html#uwsgi-setup):

```
# disable logging for privacy
disable-logging = true
```

**Reply by AlyoshaVasilieva** [reply]
I used the script to install, Debian 11, uwsgi version `2.0.19.1-debian`. `disable-logging = true` is indeed present in `/etc/uwsgi/apps-enabled/searxng.ini`. It isn't preventing the error logging, though.

**Reply by return42** [reply]
Seems Debian set some different log options other distros don't / from `/usr/share/doc/uwsgi/README.Debian.gz`

    Log of started uWSGI instance is placed at /var/log/uwsgi/app/<confname>.log

I'm not familiar with but there some other config files in Debian worth to know

    /etc/default/uwsgi
    /usr/share/uwsgi/conf/default.ini

For uWSGI debian uses the LSB init process (this might be changed one day, see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=833067 )

In the `/usr/share/uwsgi/init/specific_daemon` I see:

```
  start-stop-daemon --start --quiet \
    --pidfile "$PIDFILE" \
    --exec "$DAEMON" \
      -- --"${INHERITED_OPTION_NAME}" "${INHERITED_CONFIG}" \
         --"${CONFFILE_OPTION_NAME}" "${CONFFILE}" \
         --daemonize "/var/log/uwsgi/${CONFNAMESPACE}/${CONFNAME}.log" \
    1> /dev/null 2>&1 \
      || return 2
```

Seems debian has no option to stop logging / Sorry that I can't help more but this seems to me a very debian specific question.   May you like to ask in a debian forum if you get an answer here.
